komga/src/main/kotlin/org/gotson/komga/interfaces/rest/BookController.kt

```diff
@@ -550,6 +551,25 @@ class BookController(
     }
   }
 
+  @DeleteMapping("api/v1/books/{bookId}")
+  @PreAuthorize("hasRole('$ROLE_ADMIN')")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  fun deleteBook(
+    @PathVariable bookId: String,
+    @AuthenticationPrincipal principal: KomgaPrincipal
+  ) {
+    bookRepository.findByIdOrNull(bookId)?.let { book ->
+      if (!principal.user.canAccessBook(book)) throw ResponseStatusException(HttpStatus.FORBIDDEN)
+      if (!Files.isWritable(book.path)) {
+        logger.warn { "Cannot delete ${book.path}: Permission denied" }
+        throw ResponseStatusException(HttpStatus.CONFLICT)
```

Comment:
I would say that this type of control lies in the domain service layer. From the API standpoint, if you return NO CONTENT, then it's a command, and the command is valid if you have the rights (ie ADMIN).